include/flatbuffers/reflection.h

```diff
@@ -243,7 +255,28 @@ template<typename T> T *GetAnyFieldAddressOf(const Struct &st,
 template<typename T> bool SetField(Table *table, const reflection::Field &field,
                                    T val) {
   assert(sizeof(T) == GetTypeSize(field.type()->base_type()));
-  return table->SetField(field.offset(), val);
+  T def;
+  switch (field.type()->base_type()) {
```

Comment:
this may be quicker with an if?
if (type <= ULong) ..
else if (type <= Double) ..
else..
?